diff --git a/video/out/gpu/libmpv_gpu.c b/video/out/gpu/libmpv_gpu.c
index 1b13c79..4195299 100644
--- a/video/out/gpu/libmpv_gpu.c
+++ b/video/out/gpu/libmpv_gpu.c
@@ -2,12 +2,22 @@
 #include "hwdec.h"
 #include "libmpv_gpu.h"
 #include "mpv/render_gl.h"
+#if HAVE_METAL
+#include "mpv/render_mtl.h"
+#endif
 #include "video.h"
 #include "video/out/libmpv.h"
 
+#if HAVE_METAL
+extern const struct libmpv_gpu_context_fns libmpv_gpu_context_metal;
+#endif
+
 static const struct libmpv_gpu_context_fns *context_backends[] = {
 #if HAVE_GL
     &libmpv_gpu_context_gl,
+#endif
+#if HAVE_METAL
+    &libmpv_gpu_context_metal,
 #endif
     NULL
 };
